### PR TITLE
Move to src layout

### PR DESCRIPTION
**Pull request by adamchainz**
This has a few advantages:

* Tests are guaranteed to run against the package as installed, rather than the in-repo version. This means that any files missing from the package will be detected in tests.
* Any future packages in the repo root like `tests` won't need to be explicitly ignored in `[options.packages.find]` in `setup.cfg`.

Also just consistent with the other packages I maintain

### CODE REVIEW

**Comment by luzfcb** [comment]
did `git mv` was used to move the files? 
In my mind (because I'm not sure how effective or true my next statement is), `git mv` will force preserve git history

I had a bad time trying to investigate the history of a project that did a similar refactoring and the git history was not preserved in the process: https://github.com/Styria-Digital/django-rest-framework-jwt/pull/92

**Comment by adamchainz** [comment]
Yes. Git infers renames. Use `git log --follow` to follow renames.

File: scripts/generate_default_media_types.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from pathlib import Path
 
 module_dir = Path(__file__).parent.resolve()
-media_types_py = module_dir / ".." / "whitenoise" / "media_types.py"
+media_types_py = module_dir / "../src/whitenoise/media_types.py"
```
Comment (pauloxnet) [inline scripts/generate_default_media_types.py:11]: Can we use `parent` here instead of `..` and maintain the pathlib way to concatenate string?

Comment (adamchainz) [reply scripts/generate_default_media_types.py:11]: I believe pathlib internally translates `".."` and `"/"` to equivalent calls, regardless of OS. But if you care enough to make a PR be my guest.

**Pull request merged**